store-implementation/federated-store/src/main/java/uk/gov/gchq/gaffer/federatedstore/FederatedStore.java

```diff
@@ -328,16 +349,45 @@ public Schema getSchema(final Map<String, String> config, final User user) {
      * given csv of graphIds, with visibility of the given user.
      * </p>
      * <p>
+     * Graphs are returned once per operation, this does not allow an infinite loop of FederatedStores to occur.
+     * </p>
+     * <p>
      * if graphIdsCsv is null then all graph objects within FederatedStore
      * scope are returned.
      * </p>
      *
      * @param user        the users scope to get graphs for.
      * @param graphIdsCsv the csv of graphIds to get, null returns all graphs.
+     * @param operation   the requesting operation, graphs are returned only once per operation.
      * @return the graph collection.
      */
-    public Collection<Graph> getGraphs(final User user, final String graphIdsCsv) {
-        return graphStorage.get(user, getCleanStrings(graphIdsCsv));
+    public Collection<Graph> getGraphs(final User user, final String graphIdsCsv, final Operation operation) {
+        Collection<Graph> rtn = new ArrayList<>();
+        if (nonNull(operation)) {
+            String optionKey = FEDERATED_STORE_PROCESSED + id;
+            boolean isIdFound = !operation.getOption(optionKey, "").isEmpty();
+            if (!isIdFound) {
+                HashMap<String, String> updatedOptions = isNull(operation.getOptions()) ? new HashMap<>() : new HashMap<>(operation.getOptions());
+                updatedOptions.put(optionKey, getGraphId());
+                operation.setOptions(updatedOptions);
```

Comment:
Why not just use operation.addOption()?

Reply:
Coz I was looking for a `put()` method which wasn't there since I was dealing with properties map.